pg_search/src/postgres/mod.rs

```diff
@@ -30,6 +30,30 @@ pub mod datetime;
 pub mod types;
 pub mod utils;
 
+#[repr(u16)]
+pub enum ScanStrategy {
+    SearchConfigJson = 1,
+    TextQuery = 2,
+    SearchQueryInput = 3,
+    // NB:  Any additions here **mut** update the `amroutine.amstrategies` down below in [`bm25_handler`]
+}
+
+impl TryFrom<u16> for ScanStrategy {
+    type Error = String;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        if value == 1 {
+            Ok(ScanStrategy::SearchConfigJson)
+        } else if value == 2 {
+            Ok(ScanStrategy::TextQuery)
+        } else if value == 3 {
+            Ok(ScanStrategy::SearchQueryInput)
+        } else {
+            Err(format!("`{value}` is an unknown `ScanStrategy` number"))
+        }
+    }
+}
+
```

Comment:
Why the `u16` representation? I'm guessing you've identified a worthwhile performance optimization, and I'm curious about where else we should be using this style.

Reply:
The `pg_sys:StrategyNumber` type is a `u16` and that's the context in which these are used in our `scan.rs`.